client/src/main/scala/com/github/dapperware/slack/api/SlackPins.scala

```diff
@@ -0,0 +1,15 @@
+package com.github.dapperware.slack.api
+
+import com.github.dapperware.slack.SlackParamMagnet
+import com.github.dapperware.slack.{ SlackEnv, SlackError }
+import zio.ZIO
+
+trait SlackPins {
+  def pin(channelId: String, timeStamp: Option[String] = None): ZIO[SlackEnv, SlackError, Boolean] = {
+    val channel: (String, SlackParamMagnet) = "channel" -> channelId
+    timeStamp match {
+      case Some(value) => sendM(request("pins.add", channel, "timestamp" -> value)) >>= isOk
+      case None => sendM(request("pins.add", channel)) >>= isOk
+    }
```

Comment:
You shouldn't need the case match here `request` will flatten out the params and remove `None` values.